src/video/video_sdl2.c

```diff
@@ -273,22 +280,23 @@ bool Video_Init(int screen_magnification, VideoScaleFilter filter)
 	case FILTER_SCALE2X:
 	case FILTER_HQX:
 	default:
-		render_width = SCREEN_WIDTH * s_screen_magnification;
-		render_height = SCREEN_HEIGHT * s_screen_magnification;
+		render_width = SCREEN_WIDTH * g_screen_magnification;
+		render_height = SCREEN_HEIGHT * g_screen_magnification;
+		s_tmp_magnification = g_screen_magnification;
 	}
-	if (s_scale_filter == FILTER_SCALE2X) {
+	if (g_scale_filter == FILTER_SCALE2X) {
 		s_fullsize_buffer = malloc(render_width * render_height * sizeof(uint8));
 		if (s_fullsize_buffer == NULL) {
 			Error("Could not allocate %d bytes of memory\n", render_width * render_height * sizeof(uint8));
 			return false;
 		}
 	}
-	err = SDL_RenderSetLogicalSize(s_renderer, render_width, render_height);
+	/*err = SDL_RenderSetLogicalSize(s_renderer, render_width, render_height);
 
 	if (err != 0) {
 		Error("Could not set logical size: %s\n", SDL_GetError());
 		return false;
-	}
+	}*/
```

Comment:
No more necessary RenderSetLogicalSize, since we're using Src/Dst rectangles to RenderCopy, who better manages original aspect ratio.